src/api/tigl.cpp

```diff
@@ -4475,6 +4475,78 @@ TIGL_COMMON_EXPORT TiglReturnCode tiglIntersectWithPlane(TiglCPACSConfigurationH
     }
 }
 
+TIGL_COMMON_EXPORT TiglReturnCode tiglIntersectWithPlaneSegment(TiglCPACSConfigurationHandle cpacsHandle,
+                                                                const char*  componentUid,
+                                                                double p1x, double p1y, double p1z,
+                                                                double p2x, double p2y, double p2z,
+                                                                double wx, double wy, double wz,
+                                                                char** intersectionID)
+{
+    if (!componentUid) {
+        LOG(ERROR) << "Null pointer for argument componentUid in tiglIntersectWithPlaneSegment.";
+        return TIGL_NULL_POINTER;
+    }
+    if (!intersectionID) {
+        LOG(ERROR) << "Null pointer for argument intersectionID in tiglIntersectWithPlaneSegment.";
+        return TIGL_NULL_POINTER;
+    }
+    if (wx*wx + wy*wy + wz*wz < 1e-10) {
+        LOG(ERROR) << "Normal vector must not be zero in tiglIntersectWithPlaneSegment.";
+        return TIGL_MATH_ERROR;
+    }
+
+    // check if p1 and p2 are distinct points
+    if ( (p1x-p2x)*(p1x-p2x) + (p1y-p2y)*(p1y-p2y) + (p1z-p2z)*(p1z-p2z) < 1e-10 ) {
```

Comment:
Use CTiglPoint::distance2(const gp_CTiglPoint& other). This is much shorter and the intent is more clear.